contracts/schemes/SchemeRegistrar.sol

```diff
@@ -40,71 +40,73 @@ contract SchemeRegistrar is UniversalScheme, VotingMachineCallbacks, ProposalExe
         bytes4 permissions;
     }
 
-    // A mapping from the organization (Avatar) address to the saved data of the organization:
-    mapping(address=>mapping(bytes32=>SchemeProposal)) public organizationsProposals;
+    mapping(bytes32=>SchemeProposal) public organizationProposals;
 
-    // A mapping from hashes to parameters (use to store a particular configuration on the controller)
-    struct Parameters {
-        bytes32 voteRegisterParams;
-        bytes32 voteRemoveParams;
-        IntVoteInterface intVote;
-    }
+    IntVoteInterface public votingMachine;
+    bytes32 public voteRegisterParams;
+    bytes32 public voteRemoveParams;
+    Avatar public avatar;
 
-    mapping(bytes32=>Parameters) public parameters;
+    /**
+     * @dev initialize
+     * @param _avatar the avatar this scheme referring to.
+     * @param _votingMachine the voting machines address to
+     * @param _voteRegisterParams voting machine parameters to register scheme.
+     * @param _voteRemoveParams voting machine parameters to remove scheme.
+     */
+    function initialize(
+        Avatar _avatar,
+        IntVoteInterface _votingMachine,
+        bytes32 _voteRegisterParams,
+        bytes32 _voteRemoveParams
+    )
+    external
+    initializer
+    {
+        require(_avatar != Avatar(0), "avatar cannot be zero");
+        avatar = _avatar;
+        votingMachine = _votingMachine;
+        voteRegisterParams = _voteRegisterParams;
+        voteRemoveParams = _voteRemoveParams;
+    }
 
     /**
     * @dev execution of proposals, can only be called by the voting machine in which the vote is held.
     * @param _proposalId the ID of the voting in the voting machine
-    * @param _param a parameter of the voting result, 1 yes and 2 is no.
+    * @param _decision the voting result, 1 yes and 2 is no.
     */
-    function executeProposal(bytes32 _proposalId, int256 _param) external onlyVotingMachine(_proposalId) returns(bool) {
-        Avatar avatar = proposalsInfo[msg.sender][_proposalId].avatar;
-        SchemeProposal memory proposal = organizationsProposals[address(avatar)][_proposalId];
+    function executeProposal(bytes32 _proposalId, int256 _decision)
+    external
+    onlyVotingMachine(_proposalId)
+    returns(bool) {
+        SchemeProposal memory proposal = organizationProposals[_proposalId];
         require(proposal.scheme != address(0));
-        delete organizationsProposals[address(avatar)][_proposalId];
+        delete organizationProposals[_proposalId];
         emit ProposalDeleted(address(avatar), _proposalId);
-        if (_param == 1) {
+        if (_decision == 1) {
```

Comment:
Not exactly related, but might be nicer to have a constant for `YES` and `NO` to use here and in many other places instead of `1` and `2`.

Reply:
not related . 
please note that YES=1 NO=2 related to voting machine implementation.
any how https://github.com/daostack/arc/issues/679.